lib/ClangImporter/ImportName.cpp

```diff
@@ -1729,9 +1729,16 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
 
 /// Returns true if it is expected that the macro is ignored.
 static bool shouldIgnoreMacro(StringRef name, const clang::MacroInfo *macro) {
-  // Ignore include guards.
-  if (macro->isUsedForHeaderGuard())
-    return true;
+  // Ignore include guards. Try not to ignore definitions of useful constants,
+  // which may end up looking like include guards.
+  if (macro->isUsedForHeaderGuard()
+      && macro->getNumTokens() == 1
+      && macro->tokens()[0].isLiteral()) {
+      auto* tok = macro->tokens()[0].getLiteralData();
+      if (tok && strlen(tok) == 1 && strncmp("1", tok, 1) == 0) {
```

Comment:
Seems like this might be simpler if you wrap `tok` in a [`StringRef`](https://llvm.org/doxygen/classllvm_1_1StringRef.html)

That way you can just use `if (tok == "1")`